### PR TITLE
Karma: run on CI

### LINKED ISSUE #1721: Discovery: use Karma for integration tests
Use [Karma](https://karma-runner.github.io/5.0/index.html) for integration (and possibly unit) tests.

TODO:

- [x] Karma setup
- [x] User puppeteer for browser
- [x] Fixture implementation
- [x] Snapshotting and Percy support
- [x] Native events
- [x] Catch and fail unexpected errors
- [x] Fix debug mode
- [x] Integrate into CI, including Percy.

Karma is a popular test framework. It does in-browser testing. By default it's able to test almost against any browser, but some testing features might be available in some environments. Some key components of Karma:
 1. Karma test runner and reporter. Runs a page in a browser. Loads both code and tests in the same context in a friendly iframe.
 2. WebPack. Can reuse our development config.
 3. Jasmine (but many other alternatives, such as Mocha)
 4. Browser launchers: Chrome, Firefox, Safari. But we can also use Puppeteer.

Comparing to Jest:
 * Karma runs in a browser. Can debug at any point and see the actual editor running, inspect in DevTools, add a breakpoint in devtools or use `debugger`etc.
 * Because it's a real browser all the browser APIs are also available. E.g. window selection, IntersectionObserver, etc. But because it's a real browser, flakes happen(tm).
 * Karma uses our WebPack config mostly unmodified, so no need for library mocks: SVGR, Color, DraftJS, etc - all can run as is.
 * However, Jest does a special module loading and can mock the whole modules, default functions, etc. Karma (or rather Jasmine) can do what's possible with the WebPack module loading.
 * Both Jest and Karma would normally use synthetic events, but Karma can also be customized to use native events with Puppeteer. 
 * Some nits, such as `display.only` in Jest vs `fdisplay` in Jasmine for focused tests.
 * [Testing Library](https://testing-library.com/) extends jsdom "expect" matchers, which are convenient, e.g. `toHaveTextContent`. Jasmine doesn't have those by default (but custom matchers are also allowed).
 * Karma tests can and should also use the Testing Library, but the boundary between DOM and jsdom is not always obvious.
 * With Puppeteer screenshots are possible and natural.
 * jsdom runs each test in a clear context. But Karma runs each test suite in a separate iframe (possibly for performance reasons). That means that side-effects and global state are problematic. For instance, `styled-components` by default manage the global state (which styles have been loaded) in the global ES6-module state. It can usually be worked around, but a painful issue at times.

See this [demo](https://drive.google.com/file/d/1swMxiw_aEtnggG0EX2ixYfc_r_0ckuZz/view?usp=sharing). To try out the prototype, use the branch for #1712.

A design proposal doc for commenting can be found here:
http://doc/1kKu8QRPYNtLjhc3botd4sEIwPnUSemLXFyNPIw8Pf50#heading=h.to9an7xfy560

### PR DESCRIPTION
**Pull request by swissspidy**
## Summary

Closes #1721.

~Blocked by #1809 for the Percy part.~

## Relevant Technical Choices

Does not include coverage reporting just yet.

Makes files compatible with current Node v14 setup.

## To-do

ℹ️ For follow-up PR:

* [ ] Actually make code coverage working   
    * Right now no coverage file is generated. But there's at least a summary in the console.
    * Is `babel-plugin-istanbul` needed?

## User-facing changes

N/A

## Testing Instructions

Verify that GitHub actions workflow here is running successfully.

---

See #1721

### CODE REVIEW

File: .github/workflows/continuous-integration-karma.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            ${{ runner.os }}-npm-
+            ${{ runner.os }}-
+
+      - name: Install dependencies
+        run: npm ci
+        env:
+          CI: true
+
+      - name: Run integration tests
+        run: npm run test:karma
```
Comment (dvoytenko) [inline .github/workflows/continuous-integration-karma.yml]: Let's do `npm run test:karma -- --headless --snapshots`

**Review by dvoytenko — APPROVED** [review]
LGTM with one ask.

File: .github/workflows/continuous-integration-karma.yml (hunk 1 above)
Comment (swissspidy) [reply .github/workflows/continuous-integration-karma.yml]: Ah, was just noticing that too. Fixed!

**Comment by swissspidy** [comment]
There are some warnings but also test failures on CI:

```
Chrome Headless 83.0.4103.0 (Linux x86_64) TextEdit integration add a text edit mode should handle a commnad, exit and save FAILED
	Error: Expected false to equal true.
	    at <Jasmine>
	    at UserContext.<anonymous> (webpack:///assets/src/edit-story/elements/text/karma/edit.karma.js:107:36 <- elements/text/karma/edit.karma.js:45844:36)
	Error: Expected 'hello world!' to equal '<span style="font-weight: 700">hello world!</span>'.
	    at <Jasmine>
	    at UserContext.<anonymous> (webpack:///assets/src/edit-story/elements/text/karma/edit.karma.js:117:64 <- elements/text/karma/edit.karma.js:45851:64)
	Error: Expected 'hello world!' to equal '<span style="font-weight: 700">hello world!</span>'.
	    at <Jasmine>
	    at UserContext.<anonymous> (webpack:///assets/src/edit-story/elements/text/karma/edit.karma.js:122:33 <- elements/text/karma/edit.karma.js:45853:33)
```

**Review by dvoytenko — APPROVED** [review]
